From 0f1f916899996017da8bf1c6eb2de69f4465f711 Mon Sep 17 00:00:00 2001
From: Alastair D'Silva <alastair@d-silva.org>
Date: Tue, 6 Oct 2026 20:11:47 +1100
Subject: [PATCH 29/29] board: sunxi: Drop redundant Secure EL3 SPC and CCU SMC
 calls

The Security Permission Controller (SPC) decode ports and CCU security
switch registers are configured directly by Secure EL3 firmware (ARM
Trusted Firmware / BL31 in sunxi_security_setup()) prior to handing off
to Non-Secure EL2 U-Boot.

Issuing raw SMC calls from U-Boot proper to re-poke these secure registers
is redundant and violates Driver Model security boundaries. Remove the
redundant SMC calls and unused arm-smccc.h header include.

Signed-off-by: Alastair D'Silva <alastair@d-silva.org>
---
 board/sunxi/board.c | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/board/sunxi/board.c b/board/sunxi/board.c
index 03aa2833ca5..94c7a69635e 100644
--- a/board/sunxi/board.c
+++ b/board/sunxi/board.c
@@ -38,9 +38,6 @@
 #ifndef CONFIG_ARM64
 #include <asm/armv7.h>
 #else
-#ifndef CONFIG_SPL_BUILD
-#include <linux/arm-smccc.h>
-#endif
 #endif
 #include <asm/gpio.h>
 #include <sunxi_gpio.h>
@@ -528,18 +525,6 @@ int board_init(void)
 		return ret;
 
 #if defined(CONFIG_MACH_SUN60I_A733) && !defined(CONFIG_SPL_BUILD)
-	{
-		/* Unlock Non-Secure access to CCMU clock registers and SPC ports early */
-		struct arm_smccc_res smc_res;
-		int spc_idx;
-
-		/* CCMU_SEC_SWITCH_REG: unlock MBUS, BUS, PLL clock registers */
-		arm_smccc_smc(0xC000FF06, 0x02003F00, 0x00000007, 0, 0, 0, 0, 0, &smc_res);
-
-		/* SPC_SETTING_REGs: enable non-secure access to all 24 master ports */
-		for (spc_idx = 0; spc_idx <= 0x5C; spc_idx += 4)
-			arm_smccc_smc(0xC000FF06, 0x02054000 + spc_idx, 0xFFFFFFFF, 0, 0, 0, 0, 0, &smc_res);
-
 		/* Enable PLL_DE (0x020022E0) with output gates 0 and 1 un-gated (0xEC125600) for DE3.5 */
 		writel(0xEC125600, (void *)0x020022E0);
 
@@ -554,7 +539,6 @@ int board_init(void)
 		writel(0x80000000, (void *)0x02002A00); /* DE0_CLK_REG: DEPLL3X, div 1, gate ON */
 
 		sunxi_a733_usb_init();
-	}
 #endif
 
 	eth_init_board();
@@ -1267,17 +1251,6 @@ int board_late_init(void)
 #endif
 
 #if defined(CONFIG_MACH_SUN60I_A733) && !defined(CONFIG_SPL_BUILD)
-	/* Unlock Non-Secure access to CCMU clock registers and SPC ports */
-	struct arm_smccc_res smc_res;
-	int spc_idx;
-
-	/* CCMU_SEC_SWITCH_REG: unlock MBUS, BUS, PLL clock registers */
-	arm_smccc_smc(0xC000FF06, 0x02003F00, 0x00000007, 0, 0, 0, 0, 0, &smc_res);
-
-	/* SPC_SETTING_REGs: enable non-secure access to all 24 master ports */
-	for (spc_idx = 0; spc_idx <= 0x5C; spc_idx += 4)
-		arm_smccc_smc(0xC000FF06, 0x02054000 + spc_idx, 0xFFFFFFFF, 0, 0, 0, 0, 0, &smc_res);
-
 #if CONFIG_IS_ENABLED(DM_REGULATOR)
 	/* Ensure all critical PMIC rails are active */
 	struct udevice *reg;
-- 
2.53.0

